R/munge_unaids.R

```diff
@@ -0,0 +1,181 @@
+                      country =  !!names(.[3]))
+    )
+
+    names_cw_tt <- suppressMessages(
+      googlesheets4::read_sheet(sheet_id_names, sheet = 2) %>%
+        dplyr::filter(sheet == "HIV Test & Treat") %>%
+        dplyr::select(-sheet) %>%
+        tidyr::pivot_wider(names_from = names,
+                           values_from = names_original) %>%
+        dplyr::select(-value)
+    )
+
+    #change column names - stop if length of names is not the same as length of df
+    stopifnot(ncol(names_cw_tt) == ncol(gdrive_df))
+
+    names(gdrive_df) <- names(names_cw_tt)
+
+    gdrive_df <-  suppressWarnings(
+      gdrive_df %>%
+        dplyr::mutate(across(tidyselect:::where(is.list), ~na_if(., "NULL"))) %>%
+        dplyr::slice(-c(1,2)) %>%
+        dplyr::mutate_at(dplyr::vars(4:110), as.numeric)
+    )
+
+  }
+
+
+  gdrive_df_clean <-
+    gdrive_df %>%
+    dplyr::mutate(dplyr::across(tidyselect::contains("_"), ~gsub(" |<|>", "", .))) %>%
+    #dplyr::mutate(dplyr::across( tidyselect::contains("_"), as.numeric)) %>%
+    dplyr::mutate(regions = ifelse(country %in% regions, country, NA)) %>%
+    tidyr::fill(regions) %>% #get regions column
+    tidyr::pivot_longer(-c(year, iso, country, regions),
+                        names_to = c("indicator")) %>%
+    tidyr::separate(indicator, sep = "_", into = c("indicator", "age", "sex", "stat"))
+
+  #HIV estimates tab - specific munging
+  if (return_type == "HIV Estimates") {
+    gdrive_df_clean <- gdrive_df_clean %>%
+      dplyr::mutate(sheet = "HIV Estimates",
+                    sex = ifelse(indicator == "pmtct", "female", sex),
+                    indic_type = dplyr::case_when(
+                      indicator %in% c("prev", "incidence") ~ "percent_indics",
+                      TRUE ~ "integer_indics"
+                    )
+      )
+
+    #clean up indicator names
+    gdrive_df_clean <- gdrive_df_clean %>%
+      dplyr::mutate(indicator = dplyr::recode(indicator,
+                                              "prev" = "Prevalence",
+                                              "deaths" = "AIDS Related Deaths",
+                                              "plhiv" = "PLHIV",
+                                              "incidence" = "Incidence",
+                                              "pmtct" = "PMTCT",
+                                              "newhiv" = "New HIV Infections"))
+  }
+
+  #Test and Treat tab - specific munging
+  if (return_type == "Test & Treat") {
+    gdrive_df_clean <- gdrive_df_clean %>%
+      dplyr::mutate(sheet = "HIV Test & Treat",
+                    sex = ifelse(indicator == "pmtct", "female", sex),
+                    indic_type = dplyr::case_when(
+                      indicator %in% c("knownstatus", "plhivOnArt", "knownstatusOnArt",
+                                       "plhivVLS", "onArtVLS", "pmtctArtPct") ~ "percent_indics",
+                      TRUE ~ "integer_indics"
+                    ))
```

Comment:
combine this with the same section above for HIV estimates and then you don't need to use if statements to make the code cleaner